### PR TITLE
Simpler install process for new users

### PR DESCRIPTION
**Pull request by ianroberts**
Now we are publishing release Docker images to `ghcr.io` it is no longer strictly necessary to clone the whole gate-teamware repository in order to run a release version (or in fact a dev version) under docker-compose.  All you actually need is:

- the compose file itself
- `generate-docker-env.sh` to generate your initial `.env` file
- `create-django-db.sh` which is bind-mounted into the postgresql container to create the initial DB
- the `nginx` folder which is bind-mounted into the frontend container to reverse proxy to the back end.

Everything else comes in the images that are pulled from `ghcr.io`.

This PR provides a "quick start" mechanism to enable this compose deployment from the public images.  I've simplified the `.env` generator so that once a suitable `.env` file is in place the "production" deployment can be run purely using `docker compose up` without needing the separate `deploy.sh` script.  I've then created a "get Teamware" script that runs the user through a set of interactive prompts for the critical configuration data (admin email address, public host name, SMTP server details, etc.) and then downloads and configures a minimal compose stack with just those four items listed above, ready to be launched using `docker compose`.  This was inspired by https://acme.sh, which has a bootstrapping shell script at a short URL which you download and run to download and install the rest of the tool.

Initially I've hand-created the installation bundle but going forward the idea is that when we make a `vN.M.P` tag on GitHub the "get Teamware" script and associated install files bundle will be generated by an action and uploaded as GitHub release artifacts.  The short URL `https://gate.ac.uk/get-teamware.sh` will then become a simple redirect to the file in the _latest_ GitHub release, so anyone can install the latest version by making an empty dir and running

```
curl -LO https://gate.ac.uk/get-teamware.sh
bash ./get-teamware.sh
```

I've also moved the instructions for how to install releases out of the "developer" guide and into the main README since they're no longer specific to developers _of Teamware_ as such.

### CODE REVIEW

File: README.md
Comment (ianroberts) [inline README.md:32]: For now this is a file hosted directly on gate.ac.uk, but once we have a published release it will become a redirect to the latest GitHub release version.

File: deploy.sh
Comment (ianroberts) [inline deploy.sh:16]: The inherited process environment overrides `.env` for `docker-compose`, so we can declare `DJANGO_ALLOWED_HOSTS` directly in the `.env` but still override it if we want to for staging/test.  This then means that `./deploy.sh prod` is directly equivalent to a simple `docker compose up -d`

File: generate-docker-env.sh
Comment (ianroberts) [inline generate-docker-env.sh:18]: Asume `main` (which ultimately means the published `ghcr.io` images) if `generate-docker-env.sh` is run from outside of a git clone.

File: generate-docker-env.sh
Comment (ianroberts) [inline generate-docker-env.sh:93]: This and a number of other settings can legitimately be set to the empty string, so I've switched from `:-` (default if unset _or if set to the empty string_) to just `-` (default if unset, but if set to empty then leave empty)

File: install/get-teamware.sh
Comment (ianroberts) [inline install/get-teamware.sh:9]: This gets replaced by the release tag by the `create-release` workflow, so the script published as an artifact in a given release will install the version of Teamware corresponding to that specific release.

File: docs/docs/README.md
Comment (davidwilby) [inline docs/docs/README.md:22]: @twinkarma and I have been planning on creating a 'Deployment/Installation guide' section to the docs (i.e. `docs/docs/installguide` or similar) - this might be a good time to do that since this PR is overhauling that section anyway.

File: install/get-teamware.sh
Comment (davidwilby) [inline install/get-teamware.sh:129]: Should we include some text similar to this regarding proxy forwarding alongside the install instructions to forewarn users? Just so someone doesn't run this script and find this out right at the end?

**Review by davidwilby — APPROVED** [review]
This looks really great, I like the quickstart approach and this greatly simplifies the docker deployment.

File: install/get-teamware.sh
Comment (ianroberts) [reply install/get-teamware.sh]: I've gone one better in my latest revision - I've added an optional add-on to the `docker-compose` stack that brings up an https reverse proxy on port 443.  I don't want to make that a standard part of the stack because it only works if the machine is open to the internet on 80/443 (because it uses ACME to obtain certificates) but it's a way to mark the path of least resistance.

**Comment by ianroberts** [comment]
This PR is becoming bigger and more complex.

I noticed that we weren't passing any of the email configuration environment variables through to the backend from the compose file, so I added those; then saw we weren't even setting the "require email activation" parameter in `.env`, so added that; then realised that `check_create_superuser` would be broken if email activation is turned on, so fixed that too...  I guess since our primary deployment moved to Kubernetes the docker-compose side hasn't been maintained so much.

Anyway, I think this should be more or less ready now, and I've tested it end-to-end on an AWS EC2 machine with the Caddy HTTPS reverse proxy in the mix as well.

**Comment by ianroberts** [comment]
Caddy is a web server written in Go that is designed to be HTTPS-by-default with zero configuration.  If you tell it that it is serving a publicly-resolvable DNS hostname then it will go out and obtain trusted certificates using ACME and keep them renewed and up to date automatically.  If you tell it to serve `localhost` it can't get ACME certificates but it still generates its own certificate authority and issues server certificates from that.  The extra compose file configures Caddy in the compose stack to do essentially the same job as the ingress controller does in a Kubernetes deployment - terminate TLS and reverse proxy `/static` to the `nginx` container and everything else to the `backend`.

**Comment by ianroberts** [comment]
I'm happy if you're happy, at least to make this version 1, but since I've trodden on many of the same files as #298 I'll let you merge that first and then rebase this.

**Comment by davidwilby** [comment]
> I'm happy if you're happy, at least to make this version 1, but since I've trodden on many of the same files as #298 I'll let you merge that first and then rebase this.

If you don't mind? I was waiting for this one to be merged, happy to do it either way round.

**Comment by ianroberts** [comment]
Rebased on latest dev including privacy policy changes.  Could I ask for a final sanity check before merging as I've made a number of changes compared to the previous push - in particular I've tried to sort out proper quoting of values in the `.env` file with e80255ed8366b0af7c92af7576e40b717b2496fb but it turned out a lot more complex than I expected it to be.

**Review by davidwilby — APPROVED** [review]
Looks good to me! Sorry to keep you waiting.

**Comment by ianroberts** [comment]
Thanks, I've done a final rebase and merged into dev

**Pull request merged**
